code/modules/organs/external/_external.dm

```diff
@@ -272,42 +265,74 @@
+		return
+	//Remove internal items/organs/implants
+	if(try_remove_internal_item(W, user))
+		return
+	..()
+
+//Handles removing internal organs/implants/items still in the detached limb.
+/obj/item/organ/external/proc/try_remove_internal_item(var/obj/item/W, var/mob/user)
 	switch(stage)
 		if(0)
 			if(W.sharp)
-				user.visible_message("<span class='danger'><b>[user]</b> cuts [src] open with [W]!</span>")
+				user.visible_message(SPAN_DANGER("<b>[user]</b> cuts [src] open with [W]!"))
 				stage++
-				return
+				return TRUE
 		if(1)
 			if(istype(W))
-				user.visible_message("<span class='danger'><b>[user]</b> cracks [src] open like an egg with [W]!</span>")
+				user.visible_message(SPAN_DANGER("<b>[user]</b> cracks [src] open like an egg with [W]!"))
 				stage++
-				return
+				return TRUE
 		if(2)
 			if(W.sharp || istype(W,/obj/item/hemostat) || isWirecutter(W))
-				var/list/organs = get_contents_recursive()
-				if(LAZYLEN(organs))
-					var/obj/item/removing = pick(organs)
-					var/obj/item/organ/external/current_child = removing.loc
-
-					LAZYREMOVE(current_child.implants, removing)
-					LAZYREMOVE(current_child.internal_organs, removing)
-
-					status |= ORGAN_CUT_AWAY
-					if(istype(removing, /obj/item/organ/internal/mmi_holder))
-						var/obj/item/organ/internal/mmi_holder/O = removing
-						removing = O.transfer_and_delete()
-
-					removing.forceMove(get_turf(user))
-
-					if(user.get_empty_hand_slot())
-						user.put_in_hands(removing)
-					user.visible_message("<span class='danger'><b>[user]</b> extracts [removing] from [src] with [W]!</span>")
+				var/list/removables = get_contents_recursive()
+				if(LAZYLEN(removables))
+					var/obj/item/removing = show_radial_menu(user, src, removables, radius = 42, require_near = TRUE, use_labels = TRUE, check_locs = list(src))
+					if(removing)
+						if(istype(removing, /obj/item/organ))
+							var/obj/item/organ/O = removing
+							O.do_uninstall()
+						removing.forceMove(get_turf(user))
+
+						if(user.get_empty_hand_slot())
+							user.put_in_hands(removing)
+						user.visible_message(SPAN_DANGER("<b>[user]</b> extracts [removing] from [src] with [W]!"))
 				else
-					user.visible_message("<span class='danger'><b>[user]</b> fishes around fruitlessly in [src] with [W].</span>")
-				return
-	..()
+					user.visible_message(SPAN_DANGER("<b>[user]</b> fishes around fruitlessly in [src] with [W]."))
+				return TRUE
+	return FALSE
 
+//Handles removing child limbs from the detached limb.
+/obj/item/organ/external/proc/try_saw_off_child(var/obj/item/W, var/mob/user)
+	var/list/removables = get_limbs_recursive(TRUE)
+	if(!LAZYLEN(removables))
+		return 
+	var/obj/item/organ/external/removing = show_radial_menu(user, src, removables, radius = 42, require_near = TRUE, use_labels = TRUE, check_locs = list(src))
+	if(!istype(removing))
+		return TRUE
+
+	user.visible_message(SPAN_DANGER("<b>[user]</b> starts cutting off \the [removing] from [src] with \the [W]!"))
```

Comment:
As noted above, please either manually check for tool quality, or use `do_tool_interaction` a la
````
if(W.do_tool_interaction(TOOL_WELDER, user, src,        \
		  max(5, round((stored_max_health-stored_health) / 5)), \
		  "repairing the damage to", "repairing the damage to", \
		  "You fail to patch the damage to \the [src].",        \
		  fuel_expenditure = 1
		))
````

Reply:
Done. 
But. kinda uncomfortable to add in something that's barely used anywhere else in a bugfix tbh.
Also the defines for the return values are defined too late to be used in here. So I used literals instead for handling the no message case, and success case.